### PR TITLE
OTG Test - TE-13.1: gRIBI route ADD during Failover

### LINKED ISSUE #1329: OTG: TE-13.1: gRIBI route ADD during Failover

### PR DESCRIPTION
**Pull request by octpetre**
OTG test conversion of **_TE-13.1: gRIBI route ADD during Failover_**
Resolves #1329

### CODE REVIEW

File: feature/experimental/gribi/otg_tests/route_addition_during_failover_test/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# TE-13.1: gRIBI route ADD during Failover
```
Comment (jasdeep-hundal) [inline feature/experimental/gribi/otg_tests/route_addition_during_failover_test/README.md:1]: Please update with changes in https://github.com/openconfig/featureprofiles/commit/ad3cab762a54801c042e10699f09713d5ff867a2

File: feature/experimental/gribi/otg_tests/route_addition_during_failover_test/route_addition_during_failover_test.go [hunk 2]
```diff
@@ -0,0 +1,736 @@
...
+	nextHopCount := 63 // nextHopCount specifies number of nextHop IPs needed.
+	top.Ports().Add().SetName(atePort.ID())
+	for i := 0; i <= nextHopCount; i++ {
+		vlanID := uint16(i) + 1
+		name := fmt.Sprintf(`dst%d`, i)
+		Index := uint32(i) + 1
+		ateIPv4 := fmt.Sprintf(`198.51.100.%d`, ((4 * i) + 1))
+		dutIPv4 := fmt.Sprintf(`198.51.100.%d`, ((4 * i) + 2))
+		configureSubinterfaceDUT(t, d, dutPort, Index, vlanID, dutIPv4)
+		MAC, _ := incrementMAC(atePort1.MAC, i+1)
```
Comment (jasdeep-hundal) [inline feature/experimental/gribi/otg_tests/route_addition_during_failover_test/route_addition_during_failover_test.go:293]: Check the error/fail here if there is one.

File: feature/experimental/gribi/otg_tests/route_addition_during_failover_test/route_addition_during_failover_test.go [hunk 3]
```diff
@@ -0,0 +1,736 @@
...
+		sipv4.Enabled = ygot.Bool(true)
+	}
+	a := sipv4.GetOrCreateAddress(dutIPv4)
+	a.PrefixLength = ygot.Uint8(uint8(ipv4PrefixLen))
+}
+
+// configureATE configures a single ATE layer 3 interface.
+func configureATE(t *testing.T, top gosnappi.Config, atePort *ondatra.Port, vlanID uint16, addPort bool, Name, MAC, dutIPv4, ateIPv4 string) {
+	t.Helper()
+	if addPort {
```
Comment (jasdeep-hundal) [inline feature/experimental/gribi/otg_tests/route_addition_during_failover_test/route_addition_during_failover_test.go:334]: Configure ports outside of this function (like in the route delete test) so we don't need the check here.

File: feature/experimental/gribi/otg_tests/route_addition_during_failover_test/route_addition_during_failover_test.go [hunk 4]
```diff
@@ -0,0 +1,736 @@
...
+	ctx        context.Context
+	client     *fluent.GRIBIClient
+	dut        *ondatra.DUTDevice
+	ate        *ondatra.ATEDevice
+	top        gosnappi.Config
+	electionID gribi.Uint128
+}
+
+// createTrafficFlow generates traffic flow from source network to destination network via
+// srcEndPoint to dstEndPoint and checks for packet loss.
```
Comment (jasdeep-hundal) [inline feature/experimental/gribi/otg_tests/route_addition_during_failover_test/route_addition_during_failover_test.go:367]: Remove the check for packet loss comment?

File: feature/experimental/gribi/otg_tests/route_addition_during_failover_test/route_addition_during_failover_test.go [hunk 5]
```diff
@@ -0,0 +1,736 @@
...
+	flow.Metrics().SetEnable(true)
+	flow.TxRx().Port().SetTxName("port1").SetRxName("port2")
+	e1 := flow.Packet().Add().Ethernet()
+	e1.Src().SetValue(atePort1.MAC)
+	e1.Dst().SetChoice("value").SetValue(dstMac)
+	v4 := flow.Packet().Add().Ipv4()
+	v4.Src().SetValue(atePort1.IPv4)
+	v4.Dst().Increment().SetStart(flowArgs.flowStartAddress).SetCount(int32(flowArgs.flowCount))
+
+	ate.OTG().PushConfig(t, top)
```
Comment (jasdeep-hundal) [inline feature/experimental/gribi/otg_tests/route_addition_during_failover_test/route_addition_during_failover_test.go:382]: Move to top-level (matching the style of the route delete test)? Clarifies the intent of createTrafficFlow and then we can remove the `t` arg as well.

File: feature/experimental/gribi/otg_tests/route_addition_during_failover_test/route_addition_during_failover_test.go [hunk 6]
```diff
@@ -0,0 +1,736 @@
...
+	err = binary.Write(buf, binary.BigEndian, convMac)
+	if err != nil {
+		return "", err
+	}
+	newMac := net.HardwareAddr(buf.Bytes()[2:8])
+	return newMac.String(), nil
+}
+
+// Waits for an ARP entry to be present for ATE Port1
+func waitOTGARPEntry(t *testing.T) {
```
Comment (jasdeep-hundal) [inline feature/experimental/gribi/otg_tests/route_addition_during_failover_test/route_addition_during_failover_test.go:507]: Can we use the otgutils version?

**Review by jasdeep-hundal — CHANGES_REQUESTED** [review]

File: feature/experimental/gribi/otg_tests/route_addition_during_failover_test/README.md (hunk 1 above)
Comment (octpetre) [reply feature/experimental/gribi/otg_tests/route_addition_during_failover_test/README.md:1]: Done

File: feature/experimental/gribi/otg_tests/route_addition_during_failover_test/route_addition_during_failover_test.go (hunk 4 above)
Comment (octpetre) [reply feature/experimental/gribi/otg_tests/route_addition_during_failover_test/route_addition_during_failover_test.go:367]: Removed

File: feature/experimental/gribi/otg_tests/route_addition_during_failover_test/route_addition_during_failover_test.go (hunk 5 above)
Comment (octpetre) [reply feature/experimental/gribi/otg_tests/route_addition_during_failover_test/route_addition_during_failover_test.go:382]: Moved up

File: feature/experimental/gribi/otg_tests/route_addition_during_failover_test/route_addition_during_failover_test.go (hunk 3 above)
Comment (octpetre) [reply feature/experimental/gribi/otg_tests/route_addition_during_failover_test/route_addition_during_failover_test.go:334]: Done

File: feature/experimental/gribi/otg_tests/route_addition_during_failover_test/route_addition_during_failover_test.go (hunk 2 above)
Comment (octpetre) [reply feature/experimental/gribi/otg_tests/route_addition_during_failover_test/route_addition_during_failover_test.go:293]: Done

File: feature/experimental/gribi/otg_tests/route_addition_during_failover_test/route_addition_during_failover_test.go (hunk 6 above)
Comment (octpetre) [reply feature/experimental/gribi/otg_tests/route_addition_during_failover_test/route_addition_during_failover_test.go:507]: Yes, replaced

**Review by jasdeep-hundal — APPROVED** [review]

**Pull request merged**